packages/block-editor/src/components/copy-handler/index.js

```diff
@@ -1,77 +1,84 @@
 /**
  * WordPress dependencies
  */
+
 import { serialize, pasteHandler } from '@wordpress/blocks';
 import { documentHasSelection } from '@wordpress/dom';
-import { withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { getPasteEventData } from '../../utils/get-paste-event-data';
+import { useRef } from '@wordpress/element';
+
+function CopyHandler( { children } ) {
+	const containerRef = useRef();
+
+	const {
+		getBlocksByClientId,
+		getSelectedBlockClientIds,
+		hasMultiSelection,
+		getSettings,
+	} = useSelect( ( select ) => select( 'core/block-editor' ), [] );
+
+	const { removeBlocks, replaceBlocks } = useDispatch( 'core/block-editor' );
+
+	const {
+		__experimentalCanUserUseUnfilteredHTML: canUserUseUnfilteredHTML,
+	} = getSettings();
+
+	const handler = ( event ) => {
```

Comment:
There's one small difference that potentially can have performance impact. Since this is an inline function, potentially, this could rerender children often. I think it's good to start like you did (without any memoization) but we just need to check that the component is not being rerendered on each "store change".

For instance you can add a "console log" inside the component and type on a paragraph and see if the logging is repeated or not as you type.

Reply:
While typing console.log is not executed. When copying/pasting there are two statements in console.

Reply:
That's fine :) Thanks for the tests